packages/adblocker/src/resources.ts

```diff
@@ -21,13 +23,105 @@ function btoaPolyfill(buffer: string): string {
 }
 
 export interface Resource {
+  name: string;
+  aliases: string[];
+  body: string;
   contentType: string;
+}
+
+function isResourceValid(resource: any): resource is Resource {
+  if (resource === null) {
+    return false;
+  }
+
+  if (typeof resource !== 'object') {
+    return false;
+  }
+
+  const { name, aliases, body, contentType } = resource;
+
+  if (typeof name !== 'string') {
+    return false;
+  }
+
+  if (!Array.isArray(aliases) || !aliases.every((alias) => typeof alias === 'string')) {
+    return false;
+  }
+
+  if (typeof body !== 'string') {
+    return false;
+  }
+
+  if (typeof contentType !== 'string') {
+    return false;
+  }
+
+  return true;
+}
+
+interface Scriptlet {
+  name: string;
+  aliases: string[];
   body: string;
+  dependencies: string[];
+  executionWorld?: Scripting.ExecutionWorld;
```

Comment:
I think it's worth considering to not tie the code in `packages/adblocker` to any particular environment. That's why I commented initially. The code in this module is usually trying to be generic and not think of where it will run but now we introduce some dependency with WebExtension environment. How are the scripts different depending on environment?